salt/auth/ldap.py

```diff
@@ -283,9 +283,15 @@ def auth(username, password):
         log.error('LDAP authentication requires python-ldap module')
         return False
 
-    # If bind credentials are configured, use them instead of user's
+    # If bind credentials are configured, verify that we can a valid bind
     if _config('binddn', mandatory=False) and _config('bindpw', mandatory=False):
         bind = _bind_for_search(anonymous=_config('anonymous', mandatory=False))
+
+        # If username & password are not None, attempt to verify they are valid
+        if bind and username and password:
+            bind = _bind(username, password,
+                         anonymous=_config('auth_by_group_membership_only', mandatory=False)
+                         and _config('anonymous', mandatory=False))
```

Comment:
Is this the scenario that we had discussed with two-factor authentication or was there another scenario that you're concerned about?

Reply:
Same scenario, yes.  The OP's auth environment generates OTPs that are literally only valid once.

Reply:
@cro So far my manually testing using both CURL commands against the API and using salt commands with external authentication, function the same as they do without this change (with a valid login).  I can't see anywhere in the OP's auth environment that indicates that it's generating OTPs, can you point that out?

Reply:
I originally got that information from the discussion on #42280 and #42426 and the docs that I wrote (https://github.com/michaelgibson/salt/pull/1/files).  So the idea was that the user would be presenting a single-use password the first time, but from then on out we would just re-authenticate with the bind username and password.  Since this was happening on the CLI there wasn't a danger that another user could hijack the session after the first auth exchange.  Whether or not the OTP was used vs the bindpw was determined by the presence of `show_jid` in the payload since it was not present the first time, but all subsequent payloads contained it.